src/main/java/com/example/demo/service/ShedulerService.java

```diff
@@ -0,0 +1,67 @@
+package com.example.demo.service;
+
+import com.example.demo.model.Message;
+import com.example.demo.model.Status;
+import com.example.demo.repo.JsonRepo;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Component
+@EnableScheduling
+public class ShedulerService {
+
+    private JsonRepo jsonRepo;
+    private Set<Message> messageListDoNotSend;
+    private List<Message> messageLists;
+    private Message message;
+    private MailSender messageService;
+
+    @Autowired
+    public ShedulerService(JsonRepo jsonRepo, MailSender messageService) {
+        this.messageService = messageService;
+        this.jsonRepo = jsonRepo;
+        this.messageListDoNotSend = new LinkedHashSet<>();
+        this.messageLists = jsonRepo.loadFromJsonToList();
```

Comment:
Do not rely on the list references. Just invoke repository method in `dataCheck`. To get more abstract API you can create a method which returns you a list of upcoming messages, and put the logic of filtering inside repository

Reply:
ok. i do it